contracts/cw20-merkle-airdrop/helpers/src/airdrop.ts

```diff
@@ -0,0 +1,45 @@
+import keccak256 from 'keccak256';
+import { MerkleTree } from 'merkletreejs';
+
+class Airdrop {
+  private tree: MerkleTree;
+
+  constructor(accounts: Array<{ address: string; amount: string }>) {
+    const leaves = accounts.map((a) => keccak256(a.address + a.amount));
+    this.tree = new MerkleTree(leaves, keccak256, { sort: true });
+  }
+
+  public getMerkleRoot(): string {
+    return this.tree.getHexRoot().replace('0x', '');
+  }
+
+  public getMerkleProof(account: {
+    address: string;
+    amount: string;
+  }): string[] {
+    return this.tree
+      .getHexProof(keccak256(account.address + account.amount))
+      .map((v) => v.replace('0x', ''));
+  }
+
+  public verify(
+    proof: string[],
+    account: { address: string; amount: string }
+  ): boolean {
+    let hashBuf = keccak256(account.address + account.amount);
+
+    proof.forEach((proofElem) => {
+      const proofBuf = Buffer.from(proofElem, 'hex');
+
+      if (hashBuf < proofBuf) {
```

Comment:
I am a bit wondering on this logic, but I think I need to understand the internals of MerkleTree better first.

It doesn't really matter, but we need some test cases to ensure it matches Rust.

Rust probably does have such test cases (I will review that later).
You could write some code there to make some demo hashes and test vectors (matches and non-matches) from the Rust code.
Then write some TS tests that ensure these test vectors work in the JS code.
No need to call Rust from TS, but make sure they are the same.